src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -602,12 +602,12 @@ def update(self,
             require(access.method is AccessMethod.https, access.method)
             require(access.headers is None, access.headers)
             signed_url = access.url
-            args = signed_url.args
+            args = furl(signed_url).args
             require('X-Goog-Signature' in args, args)
             self._location = signed_url
 
     @property
-    def location(self) -> Optional[furl]:
+    def location(self) -> Optional[str]:
```

Comment:
The signature in the base class and all implementations need to be updated.

Reply:
Fixed signature in base class. The other implementations in `CannedFileDownload` and `DSSFileDownload` were never updated to `Optional[furl]`.